python-threatexchange/threatexchange/cli/match_cmd.py

```diff
@@ -198,31 +217,62 @@ def execute(self, settings: CLISettings) -> None:
                 if self.as_hashes:
                     results = _match_hashes(path, s_type, index)
                 else:
-                    results = _match_file(path, s_type, index)
+                    results = _match_file(path, s_type, index, rotations=self.rotations)
 
                 for r in results:
                     metadatas: t.List[t.Tuple[str, FetchedSignalMetadata]] = r.metadata
                     for collab, fetched_data in metadatas:
                         if not self.all and collab in seen:
                             continue
                         seen.add(collab)
+
+                        # Add rotation information if possible 
+                        rotation_info = ""
+                        if hasattr(r.similarity_info, "rotation"):
+                            rotation_info = f" [{r.similarity_info.rotation.name}]"
```

Comment:
While clever, and valid python, this will make typechecking unhappy (try running mypy locally as described in https://github.com/facebook/ThreatExchange/blob/main/python-threatexchange/CONTRIBUTING.md).

Instead, change the return type of these functions from IndexMatch to a new object that contains an index match and the rotation info.